app/ts/utils/abi.ts

```diff
@@ -81,3 +81,18 @@ export const CompoundGovernanceAbi: InterfaceAbi = [
 export const CompoundTimeLock: InterfaceAbi = [
 	'function executeTransactions(address[] memory targets, uint[] memory values, string[] memory signatures, bytes[] memory datas, uint eta) public payable'
 ]
+
+function getStringBetweenParentheses(inputString: string): string | undefined {
+	const regex = /\(([^)]+)\)/
+	const match = inputString.match(regex)
+	if (match) return match[1]
+	return undefined
+}
+
+// Transfer(address,address,uint256) -> ['address', 'address', 'uint256']
+export const extractFunctionArgumentTypes = (signature: string) => {
+	const args = getStringBetweenParentheses(signature)
+	return args === undefined ? undefined : args.split(',')
+}
```

Comment:
Are you confident you will always have the canonical function signature?  `function Transfer(address sender, address receiver, uint256 amount)` is valid (but not canonical) in many places.